### PR TITLE
IA-2322 user roles fixes front

### PR DESCRIPTION
**Pull request by beygorghor**
Multiple fixes for user roles management

Related JIRA tickets : IA-2316, IA-2317, IA-2318, IA-2319, IA-2320

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- IA-2316: remove margin top on table and margin bottom on tabs + added sticky header on permissions list
- IA-2317: pass user to the table `extraprops` to refresh state
- IA-2318: rewrite condition to map on `user_permissions` instead of `permissions` + check if has permission with user role
- IA-2319: move inputComponent to select user roles on permissions tab + refresh `user_roles_permissions` while editing `user_roles`
- IA-2320: reset dialog when save is completed => this dialog should use new `makeFullModal`

## How to test

See videos for testing

## Print screen / video

https://github.com/BLSQ/iaso/assets/12494624/493af717-e5da-4a73-8d6d-b32c79874a5f

https://github.com/BLSQ/iaso/assets/12494624/b02e68aa-a055-474e-a82b-4b910ca76cd7

### CODE REVIEW

**Comment by beygorghor** [comment]
@cypress

File: hat/assets/js/apps/Iaso/domains/userRoles/components/CreateEditUserRole.tsx [hunk 1]
```diff
@@ -114,7 +114,13 @@ export const CreateEditUserRole: FunctionComponent<Props> = ({
                 titleMessage={titleMessage}
                 onConfirm={closeDialog => {
                     handleSubmit();
-                    setCloseModal({ closeDialog });
+                    setCloseModal({
+                        closeDialog: () => {
+                            resetForm();
+                            setUserRolePermissoins([]);
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/userRoles/components/CreateEditUserRole.tsx:120]: Permissoins ??😂

File: hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx [hunk 2]
```diff
@@ -171,7 +179,9 @@ const UserDialogComponent: FunctionComponent<Props> = ({
                     />
                 </Tabs>
                 <div className={classes.root} id="user-profile-dialog">
-                    {tab === 'infos' && (
+                    <div
+                        className={tab === 'infos' ? '' : classes.hiddenOpacity}
+                    >
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx:184]: I don't understand this change, but why not. It should be extended to the locations as well then. Nothing  blocking

**Review by quang-le — APPROVED** [review]

**Review by quang-le — CHANGES_REQUESTED** [review]
When I create a new user and add a user role, the permissions are not switched:
https://github.com/BLSQ/iaso/assets/38907762/4f6c168e-6e64-4b60-8abd-bca167821f7c

**Review by quang-le — APPROVED** [review]

**Comment by beygorghor** [comment]
> When I create a new user and add a user role, the permissions are not switched: https://github.com/BLSQ/iaso/assets/38907762/4f6c168e-6e64-4b60-8abd-bca167821f7c

this is fixed already

File: hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/users/components/UsersDialog.tsx:184]: simply avoiding the tab to reload on user intiial state, but i don't think it's necessary anymore

**Review by hakifran — APPROVED** [review]
Good.

**Pull request merged**